### DISCUSSION TITLE
How I purchased my SWDICE MKII pro V2 for about 17 Euro's

### DISCUSSION THREAD

**Discussion by hmeijdam** [dbuezas/lgt8fx#101] [Show and tell]
In May this year I purchased a SWDICE MKII Pro V2 via a purchasing agent in China, after I failed to obtain this programmer directly from Taobao. I stopped when I had to open an Alipay account, after however successfully registering myself in Taobao.

Below is my programmer:
![P1130478](https://user-images.githubusercontent.com/9595465/99186006-a76f8200-274d-11eb-96b7-9bd5c00fff2b.JPG)

it's based on a Nuvoton NUC123 MCU
![P1130481](https://user-images.githubusercontent.com/9595465/99186004-a50d2800-274d-11eb-8f7a-ede9c9d17209.JPG)

I used [www.wegobuy.com](www.wegobuy.com) as an agent to buy the programmer from Taobao. 
I provided Wegobuy with this [Taobao  link](https://item.taobao.com/item.htm?spm=a1z10.1-c.w4004-2729707598.8.7dd6374cMWEc1n&id=539372088892) , so they would know what to purchase.

First I had to pay for them purchasing the item and ship it to their warehouse. This was about US$ 13
After a little while I got a notification that the item had arrived and they showed a picture of it.
Next Wegobuy determined the shipping cost to my country (Netherlands), but as they listed the programmer as a battery operated device, the shipping cost was high. I opened a case and explained it was USB powered an did not contain any batteries. This dropped the shipping cost dramatically to about USD 3.40. Combined with Wegobuy service fee of about US$ 3.50 My total cost was about US$20  (EUR 17). It took about 7 weeks to receive my programmer.

So maybe helpful for those who also want one. Make sure to add a note that it is not a battery operated device, but USB powered instead. 

Good luck,
Hans

**Comment by prosper00** [comment]
I want one. Any FOSS software work on it? What's your tool chain for debugging with it?

**Reply by LaZsolt** [reply]
Here is a document, which tells how to use SWDICE MKII with a debugger.
http://www.lgtic.com/upload/lgt8fx8p/LGT8F328P_startup_based_on_EWAVR.rar

**Comment by hmeijdam** [comment]
I only use it as a programmer in combination with the Arduino IDE core. It's somewahat similar to my nu-link II programmer (to program Nuvoton MCU's)

**Comment by dbuezas** [comment]
I'm curious, what extra functionality do you get compared to using another 328p as ISP?

**Reply by hmeijdam** [reply]
For me not much. The 328P (or LGT8F328) as ISP can now also burn the bootloeder. The only thing the SWDICE can do more is autoprogramming. If I need to burn a few MCU's the only thing I need to do is plug in the board > plug out the board. When it senses a connected board it wil automatically write the HEX file. My Nuvoton programmer has the same feature, but even better. It stores the hexfile in the programmer and therefore you only need the programmer. No connection to PC required. Maybe SWDICE has this option also somewhere, but I see no means to activate that in Logic Greens LGTMix_ISP software that manages the SWDICE when connected to my PC.

I purchased mine, before it was possible to use a 328p as ISP for LGT8F328. And my main reason was to get rid of the bootloader delay at startup. So currently I would not purchase one again.

**Comment by tistructor** [comment]
the swdio programmer for stm32 can work to program the LGT8F328 chip?
